### DISCUSSION TITLE
Support HGT for offline hillshading

### DISCUSSION THREAD

**Discussion by ztNFny** [mapsforge/vtm#1161] [Q&A] [answered]
Unlike mapsforge vtm currently supports only online-sources for hillshading (e.g. MapilionMvtActivity.java). It'd be nice to have support for HGT format hillshading information to enable offline hillshading as well

> VTM does not support currently Hillshading from digital elevation model data like [HGT](http://wiki.openstreetmap.org/wiki/SRTM#Format) format. That works in Mapsforge with a different engine, processing bitmap images ([mapsforge/mapsforge#923](https://github.com/mapsforge/mapsforge/issues/923)).

_Originally posted by @devemux86 in https://github.com/mapsforge/vtm/issues/197#issuecomment-422341553_

**Comment by devemux86** [comment]
It could work with a new offline bitmap tile source, with bitmap tiles generated from the hillshading data.
(code exists in [Mapsforge](https://github.com/mapsforge/mapsforge))

This can be combined with any existing map source, like when using online hillshading bitmap tile source.

![vtm](https://user-images.githubusercontent.com/3484020/45676724-171c5f00-bb3b-11e8-88f0-0923f8eeb18f.png)

**Comment by jhotadhari** [comment]
I copied lots of mapsforge code and build a [HillshadingTileSource](https://github.com/jhotadhari/react-native-mapsforge-vtm/blob/main/android/src/main/java/com/jhotadhari/reactnative/mapsforge/vtm/tiling/source/hills/HillshadingTileSource.java)

First idea. Maybe some improvements and it could be implemented in vtm.

**Comment by demiantres** [comment]
> I copied lots of mapsforge code and build a [HillshadingTileSource](https://github.com/jhotadhari/react-native-mapsforge-vtm/blob/main/android/src/main/java/com/jhotadhari/reactnative/mapsforge/vtm/tiling/source/hills/HillshadingTileSource.java)
> 
> First idea. Maybe some improvements and it could be implemented in vtm.

Nice work. I was also considering to leverage OpenGL ES shaders for hill shading but I have not made progress yet.

**Comment by devemux86** [comment]
@ztNFny @jhotadhari @demiantres 

VTM hillshading from HGT digital elevation model data is implemented in pull request:
- https://github.com/mapsforge/vtm/pull/1189

<img src="https://github.com/user-attachments/assets/db27b587-ae3f-422a-9ee6-060935f86d2f" width="50%"/><img src="https://github.com/user-attachments/assets/42fe9ef1-b30b-4484-b3d8-06852bd6f24d" width="50%"/>

**Comment by devemux86** [comment, marked as answer]
- Implemented by https://github.com/mapsforge/vtm/pull/1189.

**Reply by jhotadhari** [reply]
Thank you!!!
I didn't find time yet to try it out. But reading through the code , it seems a way more clean and performant then the TileSource I wrote (~wrote~ copied snippets together that were mostly written by you).

well, I guess everything I know about Java so far, I learned by reading your code ;)
